base/util.jl

```diff
@@ -441,7 +441,7 @@ will always be called.
 """
 function securezero! end
 @noinline securezero!(a::AbstractArray{<:Number}) = fill!(a, 0)
-securezero!(s::String) = unsafe_securezero!(pointer(s), sizeof(s))
+securezero!(s::Cstring) = unsafe_securezero!(pointer(s), sizeof(s))
```

Comment:
`Cstring` is an unsafe pointer type. It doesn't make sense to call `pointer` on it, and it's definitely wrong to call `sizeof` on it. The function would also need to be renamed `unsafe_securezero!`